akka-persistence-typed/src/main/scala/akka/persistence/typed/serialization/ReplicatedEventSourcingSerializer.scala

```diff
@@ -183,6 +203,11 @@ import scala.collection.immutable.TreeMap
       } else None)
   }
 
+  def taggedFromBinary(bytes: Array[Byte]): Tagged = {
+    val p = ReplicatedEventSourcing.Tagged.parseFrom(bytes)
+    Tagged(wrappedSupport.deserializePayload(p.getPayload), p.getTagsList.iterator.asScala.toSet)
+  }
+
   def counterFromBinary(bytes: Array[Byte]): Counter =
     Counter(BigInt(ReplicatedEventSourcing.Counter.parseFrom(bytes).getValue.toByteArray))
 
```

Comment:
I have a gut feeling that there is a need for unwrapping the Tagged event before passing it to the event handler on the receiving side. See onReplicatedEvent and handleExternalReplicatedEventPersist in Running.scala

Reply:
👍🏼

Reply:
hmm, it looks like replicating tagged events was already supported (https://github.com/akka/akka/blob/31f654768f86db68f4c22daa2cbd0bae28fc1fad/akka-persistence-typed-tests/src/test/scala/akka/persistence/typed/ReplicatedEventSourcingTaggingSpec.scala#L85-L115), but we just lacked a test that exercised the serialization of the messages they exchanged.

Reply:
Oh, that test is using the automatic tagging via `withTagger` not the emission of `Tagged(evt)`.

Reply:
@patriknw after reviewing some more code, I don't see how `Taggged(evt, tags)` is ever sent from one node to another when using Replicated EventSourcing. When reading events produced in other replicas from the database the tags are not read.

Reply:
Maybe the current behavior is that the tags are only added in the replica that originally emitted the event?
I wonder if that is the desired behavior or if it should also tag them in the other replicas, and maybe that should be via the `withTagger` in the other replicas rather than via serialized Tagged (since that will not be available from the DB stream).

Reply:
> Maybe the current behavior is that the tags are only added in the replica that originally emitted the event?

I'm not sure, I would have to test it out. There's some explicit code on a test that avoids tagging events when the event handling happens on a replica ( [here](https://github.com/akka/akka/blob/31f654768f86db68f4c22daa2cbd0bae28fc1fad/akka-persistence-typed-tests/src/test/scala/akka/persistence/typed/ReplicatedEventSourcingTaggingSpec.scala#L64-L70) ). Maybe not tagging replicas should be the out-of-the-box behavior.

Regarding serialization of Tagged, I don't see how it is being serialized unless the emitted event is a `Tagged(evt,tags)` instance.

I think we're digressing and may better discuss tagging edge cases in RES on a separate issue.

Reply:
> I think we're digressing and may better discuss tagging edge cases in RES on a separate issue.

I agree, please create an issue. It's unclear to me how it works and how we want it to work. Then we can bring in @johanandren and @chbatey to that discussion.

Reply:
I wonder the `Tagged` is perhaps emitted when event publishing is enabled? (https://doc.akka.io/docs/akka/current/typed/replicated-eventsourcing.html#direct-replication-of-events)

Reply:
raised https://github.com/akka/akka/issues/30091